waiter/src/waiter/process_request.clj

```diff
@@ -49,21 +49,24 @@
            (org.eclipse.jetty.server HttpChannel HttpOutput Response)))
 
 (defn check-control
```

Comment:
This `fn` was previously focused on handling issues inside the `ctrl` channel.  Does the new logic belong elsewhere?